### PR TITLE
load config from pyproject.toml

### PR DESCRIPTION
**Pull request by kotyara1005**
ConfigParser can not parse toml files

### CODE REVIEW

**Comment by SaschaKrug** [comment]
While I would really like to see real pyproject.toml support, I could also understand if the maintainers are reluctant to add another dependency. At least for our projects it would already be enough to just use add a `strict=False`, because anyway the unreadable parts of the pyproject.toml are not used by django-migration-linter. Like:
```python
config_parser = configparser.ConfigParser(strict=False)
```

**Comment by kotyara1005** [comment]
Unfortunately, even with `strict=False ` configparser can not parse toml multiline values. I could add toml as an optional dependency

Example:
```python
import configparser

pyproject_toml = """
[tool.black]
line-length = 88
target-version = ['py36', 'py37', 'py38']
include = '\.pyi?$'
extend-exclude = '''
/(
  # The following are specific to Black, you probably don't want those.
  | blib2to3
  | tests/data
  | profiling
)/
'''
"""

configparser.ConfigParser(strict=False).read_string(pyproject_toml)

```

Exception :
```
configparser.ParsingError: Source contains parsing errors: '<string>'
	[line  7]: '/(\n'
	[line 12]: ')/\n'
	[line 13]: "'''\n"
```

File: setup.py
Comment (David-Wobrock) [inline setup.py:49]: Please avoid using strict version definitions, we should prefer using lower bounds for dependencies
```suggestion
        "toml": ["toml>=0.10.2"],
```

Comment (David-Wobrock) [inline setup.py:49]: Why adding this dependency in the `extras_require`?
Do you think it makes sense to require someone to specifically install a version with `toml`?

I think we should rather always install it in order to simplify life for users

File: django_migration_linter/management/commands/lintmigrations.py
Comment (David-Wobrock) [inline django_migration_linter/management/commands/lintmigrations.py:174]: `config_parser.get(CONFIG_NAME, key, fallback=None)` is not very readable in the first place,
Let's avoid putting it twice in the same dict comprehension and abuse of this construct :)

File: django_migration_linter/management/commands/lintmigrations.py
Comment (David-Wobrock) [inline django_migration_linter/management/commands/lintmigrations.py:172]: I think we lost the `config_parser.getboolean` in the refactoring process

**Review by David-Wobrock — CHANGES_REQUESTED** [review]
Are multiline toml configurations that common?
I'd happy to please your requirements, but adding a dependency for a case that doesn't seem too common seems cumbersome for most users. And a conditional install adds useless complexity to the project 🤔

**Comment by SaschaKrug** [comment]
> Are multiline toml configurations that common?
> I'd happy to please your requirements, but adding a dependency for a case that doesn't seem too common seems cumbersome for most users. And a conditional install adds useless complexity to the project 🤔

@kotyara1005 is right. In our case, we are not using them,  but others do and its common. Pyproject.toml is currently the upcoming format for unified configurations of all python code analysis tools, so many will move all possible configurations into it and they will also use more language features of .toml. So, real .toml support would be really great.

**Comment by side2k** [comment]
We have  `pyproject.toml` with multiline statements in our project, and we can't use `django-migration-linter` in its current version just because we can't prevent it from trying to read it.
I believe, if `.toml` support is that limited - its better not to support it at all, or, at least, allow users to skip certain config files.

**Comment by SaschaKrug** [comment]
@kotyara1005 could you please have a look into the requested changes if you have time?

**Comment by David-Wobrock** [comment]
I took the liberty to rebase and push new commits. I hope it works as expected.

Some unit tests would be greatly appreciated on how config files behave in the Django management command.

If any one can test those against some toml files, that would be great - I'm not that familiar with those and a bit out of time at the moment :)

**Comment by SaschaKrug** [comment]
> If any one can test those against some toml files, that would be great - I'm not that familiar with those and a bit out of time at the moment :)
Thank you for the work. I tested it with our simple .toml, with a multiline .toml and with a faulty .toml. All are working as expected.

One thing might be worth to consider: In the pyproject.toml, it is currently convention to use subgroups. External tools like pytest, pylint or black are using the `tool` group. In case of Django-migration-linter it would be: `[tool.django_migration_linter]`
Personally I don't see any problem with it, but it might be worth to follow the convention.

Be aware: Changing this, would also require a documentation change and also would break existing configurations if someone uses already the .toml implementation. So I'm not sure if it would be worth it.

**Comment by David-Wobrock** [comment]
> 
> 
> > If any one can test those against some toml files, that would be great - I'm not that familiar with those and a bit out of time at the moment :)
> 
> Thank you for the work. I tested it with our simple .toml, with a multiline .toml and with a faulty .toml. All are working as expected.

Good news 👍  Thanks a lot for testing that out 💪 
I'll proceed to merge this once we act on the point below.

> 
> One thing might be worth to consider: In the pyproject.toml, it is currently convention to use subgroups. External tools like pytest, pylint or black are using the `tool` group. In case of Django-migration-linter it would be: `[tool.django_migration_linter]`
> Personally I don't see any problem with it, but it might be worth to follow the convention.
> 
> Be aware: Changing this, would also require a documentation change and also would break existing configurations if someone uses already the .toml implementation. So I'm not sure if it would be worth it.

If those notorious tools use this convention, let's follow it too :)
I don't mind breaking the existing toml support, as it was already buggy, especially if we document the change extensively

**Comment by David-Wobrock** [comment]
> > One thing might be worth to consider: In the pyproject.toml, it is currently convention to use subgroups. External tools like pytest, pylint or black are using the `tool` group. In case of Django-migration-linter it would be: `[tool.django_migration_linter]`
> > Personally I don't see any problem with it, but it might be worth to follow the convention.
> > Be aware: Changing this, would also require a documentation change and also would break existing configurations if someone uses already the .toml implementation. So I'm not sure if it would be worth it.
> 
> If those notorious tools use this convention, let's follow it too :)
> I don't mind breaking the existing toml support, as it was already buggy, especially if we document the change extensively

Actually, isn't that what we already do in this PR?
See https://github.com/3YOURMIND/django-migration-linter/pull/143/files#diff-3e0d0e74b61f297101a00635537882b2f3eee7eb387922ceadd82442e7435085R173

**Review by David-Wobrock — APPROVED** [review]
Pushed a rebase, let's merge for now.

Thanks all for the help 🙏

**Pull request merged**
